utils/databake/src/primitives.rs

```diff
@@ -35,6 +35,17 @@ where
     T: Bake,
 {
     fn bake(&self, ctx: &CrateEnv) -> TokenStream {
+        if core::mem::size_of::<Self>() == core::mem::size_of::<&[u8]>()
+            && core::any::type_name::<Self>() == core::any::type_name::<&[u8]>()
+        {
+            // Safety: `&T` and `&[u8]` have the same size. Note that `self: &&T`, so this copies
```

Comment:
Question: Why do we need this impl on `&&[u8]` in the first place; why aren't we sticking to just the `&[u8]` impl, which is much easier to reason about?

Reply:
because then every single byte literal will be `&*b"foo"`.

Reply:
This is the `&[u8]` impl, below is the `[u8]` impl (the `&self` values in these impls are `&&[u8]` and `&[u8]`). We have to impl for `[T]` instead of `&[T]` because we want to impl on all references and don't have specialization.

Reply:
What I mean is that I think the Bake _impl_ on `&'a T` where `T = [u8]` is just a convenience to allow borrowed fields of structs to work nicer, but since we have custom Bake impls on the ZeroVec types we care about, we don't really need the borrow impl since we can deref and call the unsized impl.

But I don't have a strong opinion and am happy with what you wrote so long as @Manishearth  is happy.

Reply:
Derefing to call another implementation is annoying with databake, because then you get an expression that produces the derefed type, i.e. `*b"foo"`. Then you have to borrow that again and get `&*b"foo"`. This implementation is to avoid the `&*`.